flytekit/types/structured/snowflake.py

```diff
@@ -0,0 +1,105 @@
+import re
+import typing
+
+import pandas as pd
+import snowflake.connector
+from snowflake.connector.pandas_tools import write_pandas
+
+import flytekit
+from flytekit import FlyteContext
+from flytekit.models import literals
+from flytekit.models.types import StructuredDatasetType
+from flytekit.types.structured.structured_dataset import (
+    StructuredDataset,
+    StructuredDatasetDecoder,
+    StructuredDatasetEncoder,
+    StructuredDatasetMetadata,
+)
+
+SNOWFLAKE = "snowflake"
+
+
+def get_private_key() -> bytes:
+    from cryptography.hazmat.backends import default_backend
+    from cryptography.hazmat.primitives import serialization
+
+    pk_string = flytekit.current_context().secrets.get(None, "snowflake", encode_mode="r")
+    # cryptography needs str to be stripped and converted to bytes
+    pk_string = pk_string.strip().encode()
+    p_key = serialization.load_pem_private_key(pk_string, password=None, backend=default_backend())
+
+    pkb = p_key.private_bytes(
+        encoding=serialization.Encoding.DER,
+        format=serialization.PrivateFormat.PKCS8,
+        encryption_algorithm=serialization.NoEncryption(),
+    )
+
+    return pkb
+
+
+def _write_to_sf(structured_dataset: StructuredDataset):
+    if structured_dataset.uri is None:
+        raise ValueError("structured_dataset.uri cannot be None.")
+
+    uri = structured_dataset.uri
+    _, user, account, warehouse, database, schema, table = re.split("\\/|://|:", uri)
```

Comment:
this looks very brittle. Does the uri always come like this?

Reply:
We create by this format, so it's 100% worked